proxmox/nodes/vms/custom_amdsev.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package vms
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/bpg/terraform-provider-proxmox/proxmox/types"
+)
+
+// CustomAMDSEV handles AMDSEV parameters.
+type CustomAMDSEV struct {
+	Type         *string           `json:"type"           url:"type"`
```

Comment:
Minor, I think `Type` is mandatory, so pointer shouldn't be used, per convention.